src/main/java/Task.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * The Task class is an abstraction for representing tasks in real life,
+ * where a task could either be done or not done.
+ *
+ * @author Chai Ming How
+ * @since 2024-02-01
+ */
+
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    public abstract String getTaskIcon();
+    public void markAsDone() {
+        isDone = true;
+    }
+
+    public void markAsNotDone() {
+        isDone = false;
+    }
+
+    public String getDescription(){
+        return this.description;
+    }
```

Comment:
It is nice how there was consistency in the coding standards, maybe you could add a spacing here?